### PR TITLE
[GPU] Add new operation GatherND-8

### PR DESCRIPTION
**Pull request by kelvinchoi-intel**
Signed-off-by: Kelvin Choi <kelvin.choi@intel.com>

### Details:
 - *Add new operation GatherND v8 for GPU*

### Tickets:
 - *CVS-66656*

### CODE REVIEW

File: inference-engine/thirdparty/clDNN/api/cldnn/primitives/gather_nd.hpp [hunk 1]
```diff
@@ -19,25 +19,37 @@ struct gather_nd : public primitive_base<gather_nd> {
...
+    /// @param batch_merged_output  batched output shape is merged as a dimention for v5.
+    ///                             In case of output{3, 2, 4, 5} at batch_dims = 2, real output shape should be {6, 4, 5}.
+    ///                             This should be false for v8.
+    ///                             For batch_dims < 2, This doesn't have any meaning.
     gather_nd(const primitive_id& id,
               const primitive_id& data,
               const primitive_id& indices,
               const uint8_t indices_rank,
               const uint8_t batch_dims = 0,
+              const bool batch_merged_output = true,
```
Comment (yeonbok) [inline inference-engine/thirdparty/clDNN/api/cldnn/primitives/gather_nd.hpp:37]: I think the newly added argument are better to be located at the last position for the backward comppatibility

**Comment by yeonbok** [comment]
LGTM with one minor comment

**Review by yeonbok — APPROVED** [review]

File: inference-engine/thirdparty/clDNN/api/cldnn/primitives/gather_nd.hpp (hunk 1 above)
Comment (kelvinchoi-intel) [reply inference-engine/thirdparty/clDNN/api/cldnn/primitives/gather_nd.hpp:37]: Your suggestion is reasonable. However, almost all other primitives have the params, 'ext_prim_id' and 'output_padding' at the last position. I think this consistency is more important.

**Review by vladimir-paramuzov — APPROVED** [review]
LGTM. Please rebase and resolve conflicts

**Pull request merged**